sift/src/main/java/siftscience/android/DevicePropertiesCollector.java

```diff
@@ -183,8 +184,10 @@ private AndroidDevicePropertiesJson get() {
         for (String packageName : packages) {
             try {
                 // Root app detected
-                pm.getPackageInfo(packageName, 0);
-                packagesFound.add(packageName);
+                if (pm != null) {
+                    pm.getPackageInfo(packageName, 0);
+                    packagesFound.add(packageName);
```

Comment:
does it this line really depend on whether pm references is null or not?

Reply:
Yes, if the package manager is null it will produce the NPE at _getPackageInfo_().
In this line, the pm will check whether the provided package name is installed on the system. It will throw _NameNotFoundException_ if the given package cannot be found on the system and hence the package name should not be added to the list.